Forms/Form.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DUCK.Forms.Fields;
+using DUCK.Utils;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DUCK.Forms
+{
+	public class Form : MonoBehaviour
+	{
+		[SerializeField]
+		private Button submitButton;
+		[SerializeField]
+		private List<FieldConfig> fieldConfigs;
+
+		public event Action<Dictionary<string, AbstractFormField>> OnSubmitForm;
+
+		private void Awake()
+		{
+			if (submitButton != null)
+			{
+				SetSubmitButton(submitButton);
+			}
+		}
+
+		/// <summary>
+		/// Validate the fields and if successful invoke the OnSubmitForm event.
+		/// Also handles clearing fields on submission if applicable.
+		/// </summary>
+		public void SubmitForm()
+		{
+			var isValid = true;
+
+			foreach (var fieldConfig in fieldConfigs)
```

Comment:
Can't you combine these two foreach loops? I get that you need the data to validate first before it can be cleared but can't you just clear the field anyway after validation is done?

Reply:
No  Fields need to be validated before the fields get cleared. This is because vildation components may require values from other fields.

Adding a comment